src/main/java/com/alipay/oceanbase/rpc/location/LocationUtil.java

```diff
@@ -660,6 +667,81 @@ public static TableEntry getTableEntryLocationFromRemote(Connection connection,
         return tableEntry;
     }
 
+    public static Long getTableIdFromRemote(ObServerAddr obServerAddr, ObUserAuth sysUA,
+                                            long connectTimeout, long socketTimeout, String tenantName,
+                                            String databaseName, String tableName) throws ObTableEntryRefreshException {
+        Long tableId = null;
+        Connection connection = null;
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        try {
+            String url = formatObServerUrl(obServerAddr, connectTimeout, socketTimeout);
+            connection = getMetaRefreshConnection(url, sysUA);
+            ps = connection.prepareStatement(PROXY_TABLE_ID_SQL);
+            ps.setString(1, tenantName);
+            ps.setString(2, databaseName);
+            ps.setString(3, tableName);
+            rs = ps.executeQuery();
+            if (rs.next()) {
+                tableId = rs.getLong("table_id");
+            } else {
+                tableId = 0L;
+            }
+        } catch (Exception e) {
+            throw new ObTableEntryRefreshException("fail to get " + tableName
+                    + " table_id from remote", e);
+        } finally {
+            try {
+                if (null != rs) {
+                    rs.close();
+                }
+                if (null != ps) {
+                    ps.close();
+                }
+            } catch (SQLException e) {
+                // ignore
+            }
+        }
+        return tableId;
+    }
+
+    public static ObIndexInfo getIndexInfoFromRemote(ObServerAddr obServerAddr, ObUserAuth sysUA,
+                                                     long connectTimeout, long socketTimeout,
+                                                     String indexTableName)
+            throws ObTableEntryRefreshException {
```

Comment:
If we use another exception when the query result is empty, try using `Exception`.

Reply:
I try to use `ObTableEntryRefreshException` since the logic is similar to fetch table entry